jib-core/src/main/java/com/google/cloud/tools/jib/cache/Caches.java

```diff
@@ -107,22 +106,31 @@ public static Initializer newInitializer(Path applicationCacheDirectory) {
 
   /** Instantiate with {@link Initializer#init}. */
   private Caches(Path baseCacheDirectory, Path applicationCacheDirectory)
-      throws CacheMetadataCorruptedException, NotDirectoryException {
-    baseCache = Cache.init(baseCacheDirectory);
+      throws CacheMetadataCorruptedException, IOException {
     applicationCache = Cache.init(applicationCacheDirectory);
-  }
 
-  @Override
-  public void close() throws IOException {
-    baseCache.close();
-    applicationCache.close();
+    // Ensures that only one Cache is initialized if using the same directory.
+    if (Files.isSameFile(baseCacheDirectory, applicationCacheDirectory)) {
```

Comment:
just curious, does this need to be syncrhonized or made threadsafe somehow?

Reply:
Yes, most things in this code are designed to be thread-local and/or immutable but the `CacheMetadata` in a Cache is shared between threads. The only layer that mutates the `CacheMetadata` state is `CacheMetadata#addLayer`, which is `synchronized`.